packages/edit-site/src/components/async/index.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+import { createQueue } from '@wordpress/priority-queue';
+
+const blockPreviewQueue = createQueue();
+
+/**
+ * Renders a component at the next idle time.
+ * @param {*} props
+ */
+export function Async( { children, placeholder } ) {
```

Comment:
Do we expect this component to be used by any edit-site component? If we don't, we may want to make it even more private by embedding it as part of DataViews.

This way, we could explore (not necessarily in this PR) making it a flag of the field API. For example:

```js
const fields = [
  {
    id: 'preview',
    render: () => {},
    isAsync: true /* This would tell DataViews to wrap the render component with Async */
  },
  { ... }
];
```

Reply:
`isAsync` would be a public API, no? The async component is already private? Additionally I like the component because you can make things async on a granular level. Normally fields shouldn't be async, the block preview seems like a rare case.

Reply:
Sounds good to me. I was thinking that providing a `isAsync` for 3rd parties (plugins, etc.) would absorb certain complexity for them. But no need to do this now.